packages/yarnpkg-shell/sources/pipe.ts

```diff
@@ -53,6 +53,9 @@ export function makeProcess(name: string, args: Array<string>, opts: ShellOption
       stderr,
     ]});
 
+    const sigtermHandler = () => child.kill(`SIGTERM`);
+    process.on(`SIGTERM`, sigtermHandler);
```

Comment:
Isn't adding all these SIGTERM handlers goign to make node complain?

Reply:
It might with `yarn workspaces foreach`, yep. @ganemone can you use a similar refcount to register a single callback, and keep track of the children manually in a set?

Reply:
it is possible if you are spawning many concurrent things, but we do remove the handler when the child process closes. We could alternatively keep track of active children in a Set in parent scope and use a single global listener.

Reply:
Do we need to explicitly forward the SIGINT as well? :thinking:

Reply:
No, iirc SIGINT is purposefully ignored because the shell already sends it to the whole process tree, so we exit once the children have caught it and exited cleanly. I think there are discussions about that (perhaps on Discord).

Reply:
FWIW if you try to run `kill -s SIGINT <yarn_pid>` nothing happens. Should it not be forwarded to the child process?

Reply:
I'm not sure under which circumstances yarn itself might receive a SIGINT though. I can't think of any, but I'm not sure why forwarding it should be prevented.

Babel for example forwards it to itself: https://github.com/babel/babel/blob/master/packages/babel-node/src/babel-node.js#L99

Reply:
The *shell* sends it to the whole process tree. Using `kill` doesn't. More details [here](https://github.com/yarnpkg/berry/issues/991#issuecomment-627947599).

In any case, this is intentional, SIGINT is working as expected. Babel doesn't have to work with many types of binaries, some of them behaving weirdly if they receive the same signal multiple times 🙂